conans/cli/api/subapi/install.py

```diff
@@ -40,13 +42,64 @@ def install_consumer(deps_graph, generators=None, source_folder=None, output_fol
         conanfile.folders.set_base_generators(output_folder)
         conanfile.folders.set_base_build(output_folder)
 
-        if deploy:
-            mod, _ = load_python_file(deploy)
-            mod.deploy(conanfile, output_folder)
+        _do_deploys(self.conan_api, conanfile, deploy, output_folder)
 
         # Add cli -g generators
         conanfile.generators = list(set(conanfile.generators).union(generators or []))
         write_generators(conanfile)
 
         if type(conanfile).system_requirements != ConanFile.system_requirements:
             call_system_requirements(conanfile)
+
+
+def _do_deploys(conan_api, conanfile, deploy, output_folder):
+    # Handle the deploys
+    cwd = os.getcwd()
+    for d in deploy or []:
+        # First, find the file containing the deployer, could be local or in the cache
+        if d.startswith("conan"):  # built-in!
+            deployer = {"conan_full_deploy": conan_full_deploy}[d]
```

Comment:
I don't understand why that `startswith` and the dictionary, why not just check with `d=="conan_full_deploy"`?

Reply:
I wanted to have an explicit way of differentiating a built-in one of an external one, and reserve the name prefix ``conan_`` for it, with the current implementation it will fail for any ``--deploy=conan_xxxx`` if users decide to name ``conan_xxxx`` to their own deployers.

Reply:
Yes, creating a deployer which name is exactly the same as the builtin is something a user should not do... 
The thing is that for other extensions/templates we don't add the conan prefix to reserve, like for example using `conan_cmake_lib` or `conan_meson_exe` for `conan new`... if we are reserving that prefix for builtins, maybe we should do for everything, but if we have a clear, documented order of resolution I don't think it's necessary.
And you can always protect your own deployers naming them with `user_deploy_full.py`